packages/ui/src/table/TableHeaderCell.tsx

```diff
@@ -12,58 +16,78 @@ export const TableHeaderCell = ({
   children,
   index,
   sortable,
+  align = "left",
+  title,
   onSortChange,
 }: TableHeaderCellProps): ReactElement => {
+  const {theme} = useTheme();
+
   const {columns, setSortColumn, sortColumn} = useTableContext();
   const width = columns[index];
   if (!width) {
     console.warn(`No width defined for column ${index} in TableHeaderCell`);
   }
+  if (children && title) {
+    console.warn("Both children and title are defined in TableHeaderCell. Title will be ignored.");
+  }
+  if (!children && !title) {
+    console.error("Either children or title is required in TableHeaderCell");
+  }
+
+  const sort = sortColumn?.column === index ? sortColumn.direction : undefined;
+  let alignItems: AlignItems = "start";
+  if (align === "center") {
+    alignItems = "center";
+  } else if (align === "right") {
+    alignItems = "end";
+  }
 
-  const onClick = () => {
+  const onClick = useCallback(() => {
     // desc => asc => undefined
     const newSort = sort === "desc" ? "asc" : sort === "asc" ? undefined : "desc";
     if (setSortColumn) {
       setSortColumn(newSort ? {column: index, direction: newSort} : undefined);
     }
     onSortChange && onSortChange(newSort);
-  };
-  const sort = sortColumn?.column === index ? sortColumn.direction : undefined;
+  }, [index, onSortChange, setSortColumn, sort]);
 
-  if (sortable) {
-    if (!onSortChange) {
-      console.error("onSortChange is required when sortable is true");
-    }
-    return (
-      <Box
-        alignItems="center"
-        direction="row"
-        flex="grow"
-        justifyContent="between"
-        marginBottom={2}
-        marginTop={2}
-        maxWidth={width}
-        minHeight={36}
-        width={width}
-        onClick={onClick}
-      >
-        {children}
-        {Boolean(sort) && (
-          <Box paddingX={2}>
-            <IconButton
-              accessibilityLabel="sort"
-              iconName={sort === "asc" ? "arrow-down" : "arrow-up"}
-              onClick={() => {}}
-            />
-          </Box>
-        )}
-      </Box>
-    );
-  } else {
-    return (
-      <Box flex="grow" justifyContent="center" marginBottom={2} marginTop={2} width={width}>
-        {children}
-      </Box>
-    );
+  if (sortable && !onSortChange) {
+    console.error("onSortChange is required when sortable is true");
   }
+  return (
+    <Box
+      // accessibilityHint="press to change sorting alphabetical order"
```

Comment:
I'm guessing this is waiting on the BoxProps change you made for accessibility?

Reply:
Oh yes! Updated!